projects/plugins/jetpack/modules/seo-tools/class-jetpack-seo.php

```diff
@@ -57,6 +57,47 @@ public function init() {
 		Jetpack_SEO_Posts::register_post_meta();
 		// Exclude posts with 'jetpack_seo_noindex' set true from the Jetpack sitemap.
 		add_filter( 'jetpack_sitemap_skip_post', array( 'Jetpack_SEO_Posts', 'exclude_noindex_posts_from_jetpack_sitemap' ), 10, 2 );
+		add_action( 'rest_api_init', array( $this, 'add_custom_field_post_type_meta' ) );
+	}
+
+	/**
+	 * Add custom field meta to all public post types that don't already have it.
+	 */
+	public function add_custom_field_post_type_meta() {
+		/**
+		 * Filter the list of post types for which custom fields support is added.
+		 *
+		 * This filter allows modification of the post types that will be processed
+		 * to add support for custom fields if they do not already support it.
+		 *
+		 * @since $$next-version$$
+		 *
+		 * @param array $post_types An array of post type names.
+		 */
+		$post_types = apply_filters(
+			'jetpack_seo_custom_field_post_types',
+			get_post_types(
+				array(
+					'public'   => true,
+					'show_ui'  => true,
+					'_builtin' => false,
+				)
+			)
+		);
+
+		// Filter out post types that already support 'custom-fields'.
+		$unsupported_post_types = array_filter(
+			$post_types,
+			function ( $post_type ) {
+				return ! post_type_supports( $post_type, 'custom-fields' );
+			}
+		);
+
+		foreach ( $unsupported_post_types as $post_type ) {
+			if ( ! post_type_supports( $post_type, 'custom-fields' ) ) {
+				add_post_type_support( $post_type, 'custom-fields' );
+			}
+		}
```

Comment:
Could this part be simplified like this:
```suggestion
    foreach ( $post_types as $post_type ) {
        if ( ! post_type_supports( $post_type, 'custom-fields' ) ) {
            add_post_type_support( $post_type, 'custom-fields' );
        }
    }
```

Reply:
Good call.
I think keeping `array_filter` is only beneficial performance-wise with a much larger array of post types. For most cases it should be fine to simplify - though keeping the additional check in, from what I understand, makes negligible difference for performance and memory.
It is simpler to read though without the `array_filter`. I'll make the change - I'll add a separate commit instead of commit your suggestion though as I think the linter will complain about the spaces :)

Reply:
:old_man_yells_at_spaces_vs_tabs